modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/consistency/GridNearReadRepairAbstractFuture.java

```diff
@@ -254,51 +275,114 @@ else if (!canRemap)
      */
     protected abstract void reduce();
 
+    /**
+     *
+     */
+    protected Map<KeyCacheObject, EntryGetResult> check() throws IgniteCheckedException {
```

Comment:
Please, mark all protected methods with `final`, also please add some javadocs here, at least description of returning value. This method is important and it should be marked as important somehow, because CHECK_ONLY strategy relies on this method only.